add_demo_src/add_demo.py

```diff
@@ -0,0 +1,36 @@
+from exceptions import DuplicateNameOrKeyException
+
+import json
+
+def add(new_demo):
+    try:
+        check_for_duplicates(new_demo.name, new_demo.key)
+        insert_demo(new_demo)
+    except DuplicateNameOrKeyException as e:
+        print(f"DuplicateNameOrKeyException: {e.message}")
+
+    return
+
+def get_existing_demos():
+    with open("../demo_list.json") as json_file:
+        data = json.load(json_file)
+        
+        return data
+
+def check_for_duplicates(name, key):
+    demos = get_existing_demos()
+    print(f"demos from file: {demos}")
+
+    for demo in demos:
+        if demo['name'] == name or demo['key'] == key:
+            raise DuplicateNameOrKeyException
```

Comment:
This gets me to think that there is maybe no sense in having both demo names and keys at all. There is nothing that emunittest uses these for any internal/special purpose - originally demo key was to distinguish duplicates, and demo name was the user-visible representation.

But it would be simpler to just have demo key be equal to demo name (which, effectively, means not having a separate demo key at all).

Reply:
I could see someone wanting to make the key something like `game-build-a` and the name something like `Game Build A` - but I agree that its not necessary. If we are going to have a name it probably shouldn't have to be unique.

I think that change is out of the scope of this PR, though